### PR TITLE
Series

### PR DESCRIPTION
**Pull request by StevenRoot**
Initial problem implementation

### CODE REVIEW

File: exercises/series/src/series.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// inputs:
+//    inputText - bounded to MAX_INPUT_TEXT_LENGTH
+//    substringLength - the desired length of substrings that are returned bounded to MAX_SERIES_LENGTH
+//
+// outputs:
+//    SeriesResult_t results - structure containing count of the substring and an array of
+//       pointers to the substrings.
+//
+//       calling routine will free memory associated with each of the substrings and
+//       the resutls structure.
```
Comment (ryanplusplus) [inline exercises/series/src/series.h]: `resutls` => `results

File: exercises/series/test/test_series.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                     actualSolution->substringCount);
+
+   // test each substring...
+   for (unsigned int index = 0; index < expectedSolution->substringCount;
+        index++) {
+      TEST_ASSERT_EQUAL_STRING(expectedSolution->substring[index],
+                               actualSolution->substring[index]);
+      free(actualSolution->substring[index]);
+   }
+   free(actualSolution->substring);
```
Comment (ryanplusplus) [inline exercises/series/test/test_series.c]: Does this free un-malloced memory when the substring count or the substring length are 0?

File: exercises/series/src/example.c [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "series.h"
+
+SeriesResults_t *series(char *inputText, unsigned int substringLength)
+{
+   int substringCount = strlen(inputText) - (substringLength - 1);
+   SeriesResults_t *results = malloc(sizeof(SeriesResults_t));
```
Comment (ryanplusplus) [inline exercises/series/src/example.c]: I don't think it's really necessary to `malloc` this structure. It is fixed size and so can be returned by value.

**Comment by StevenRoot** [comment]
Initial comments addressed.

**Comment by ryanplusplus** [comment]
Looks good to me

**Comment by patricksjackson** [comment]
LGTM 👍

**Pull request merged**